File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.cpp
class Solution {
public:
    bool isVowel(char c) {
        return (c=='a' || c=='e' || c=='i' || c=='o' || c=='u' || c=='A' || c=='E' || c=='I' || c=='O' || c=='U');
    }

    string sortVowels(string s) {
        int n = s.size();
        vector<char> v;

        for(int i=0; i<n; i++) {
            if(isVowel(s[i]))   v.push_back(s[i]);
        }

        sort(v.begin(), v.end());

        int vIndex = 0;
        string t = "";
        for(int i=0; i<n; i++) {
            if(isVowel(s[i]))   t += v[vIndex++];
            else    t += s[i];
        }

        return t;
    }
};

File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.java
class Solution {
    private boolean isVowel(char c) {
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ||
           c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U')
            return true;
        
        return false;
    }

    public String sortVowels(String s) {
        char[] arr = s.toCharArray();
        ArrayList<Character> vowels = new ArrayList<>();

        for(char c : arr) {
            if(isVowel(c)) {
                vowels.add(c);
            }
        }

        Collections.sort(vowels);
        int index = 0;

        for(int i=0; i<arr.length; i++) {
            if(isVowel(arr[i])) {
                arr[i] = vowels.get(index);
                index++;
            }
        }

        return new String(arr);
    }
}